assets/js/extensions/shipping-methods/pickup-location/save.tsx

```diff
@@ -26,7 +26,15 @@ const SaveSettings = () => {
 				variant="primary"
 				isBusy={ isSaving }
 				disabled={ isSaving }
-				onClick={ save }
+				onClick={ (
+					event: React.MouseEvent< HTMLButtonElement, MouseEvent >
+				) => {
+					const target = event.target as HTMLButtonElement;
+					if ( target?.form?.reportValidity() ) {
+						save();
+					}
+				} }
+				type="submit"
```

Comment:
This is all that's needed to show errors, `reportValidity` will return true once all fields are valid.